cpp/arcticdb/storage/storage.hpp

```diff
@@ -80,34 +80,39 @@ class KeyNotFoundException : public ArcticSpecificException<ErrorCode::E_DUPLICA
     mutable std::string msg_;
 };
 
-template<class Impl>
-class Storage : public StorageBase {
+class Storage {
 public:
 
     Storage(LibraryPath library_path, OpenMode mode) :
         lib_path_(std::move(library_path)),
         mode_(mode) {}
 
+    virtual ~Storage() = default;
+    Storage(const Storage&) = delete;
+    Storage& operator=(const Storage&) = delete;
+    Storage(Storage&&) = default;
+    Storage& operator=(Storage&&) = delete;
```

Comment:
Note to reviewers: this does not entirely respect entity semantic (the move constructor is not deleted), but this is coherent with some other existing class declarations and definitions.